### ISSUE TITLE
twisted.mail.test.test_mail.MaildirAppendStringTestCase.testAppend fails intermittently on some builders

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4467]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#4467|
|Type|defect|
|Created|2010-05-30 18:25:19Z|
|Branch|https://github.com/twisted/twisted/tree/reliable-appendMessage-4467|

For example, [this build](http://buildbot.twistedmatrix.com/builders/hardy32-py2.5-glib2/builds/95) failed because of it.

Possibly this is related to [#3812](https://github.com/twisted/twisted/issues/3812)?

<details><summary>Searchable metadata</summary>

```
trac-id__4467 4467
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_reliable_appendMessage_4467 branches/reliable-appendMessage-4467
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1275243919000000 1275243919000000
changetime__1275345609000000 1275345609000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [29108]) Branching to 'reliable-appendMessage-4467'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

The problem is that message ordering is defined by when delivery completes, not when it starts.  The test assumed that delivery would complete in the same order it was started, but on a machine with a clock of low enough precision this stopped being true.

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/reliable-appendMessage-4467)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @glyph<br>@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @exarkun<br>@glyph set status to `new`|
|-|-|

OK, this looks good to me.  Some minor stuff to fix, if you feel like it:

  1. It would be nice if `appendMessage` had a `@return` or something.  It's really not clear to me when that deferred is supposed to fire, and it doesn't seem to be part of an interface.
  1. Pyflakes reports the imports from the top of the file as unused.  (Actually, nevermind.  Maybe just file a separate easy ticket for this one, since it seems to be over the file and unrelated to your changes.)
  1. Shouldn't `_AppendTestMixin` inherit from `object`, since it's a new class?

This looks like a big improvement though, so, merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [29148]) Expand docstring for appendMessage

refs [#4467](https://github.com/twisted/twisted/issues/4467)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [29149]) Make _AppendTextMixin new-style

refs [#4467](https://github.com/twisted/twisted/issues/4467)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [29150]) Merge reliable-appendMessage-4467

Author: exarkun
Reviewer: glyph
Fixes: [#4467](https://github.com/twisted/twisted/issues/4467)

Change the tests for `MaildirMailbox.appendMessage` so that they only assert
a deterministic delivery-completion based delivery order.  Previously they
attempted to assert delivery-initiation based order, which is not guaranteed.
Also remove the `callLater(0, ...)` in the implementation of `appendMessage`,
as it really is not necessary.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
